### PR TITLE
feat: allowing reactor formulas to be updated

### PR DESCRIPTION
**Pull request by armsteadj1**
## Description

- allowing reactor formulas

## Testing required outside of automated testing?

- [ ] Not Applicable

### Screenshots (if appropriate):

- [ ] Not Applicable

## Rollback / Rollforward Procedure

- [ ] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: src/commands/reactorFormulas/update.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  REACTOR_FORMULA_FLAGS,
+} from '../../reactorFormulas/utils';
+
+export default class Update extends BaseCommand {
+  public static description =
+    'Updates an existing Reactor Formula. Requires `reactor:update` Management Application permission';
+
+  public static examples = [
+    '<%= config.bin %> <%= command.id %> 03858bf5-32d3-4a2e-b74b-daeea0883bca',
+    '<%= config.bin %> <%= command.id %> 03858bf5-32d3-4a2e-b74b-daeea0883bca --code ./myRequestTransform.js',
```
Comment (djejaquino) [inline src/commands/reactorFormulas/update.ts]:
```suggestion
    '<%= config.bin %> <%= command.id %> 03858bf5-32d3-4a2e-b74b-daeea0883bca --code ./formula.js',
```

File: src/commands/reactorFormulas/update.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    '<%= config.bin %> <%= command.id %> 03858bf5-32d3-4a2e-b74b-daeea0883bca --code ./myRequestTransform.js -w',
```
Comment (djejaquino) [inline src/commands/reactorFormulas/update.ts]:
```suggestion
    '<%= config.bin %> <%= command.id %> 03858bf5-32d3-4a2e-b74b-daeea0883bca --code ./formula.js -w',
```

File: src/reactorFormulas/management.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+const debug = require('debug')('reactor-formulas:management');
+
+const listReactorFormulas = async (
+  bt: IBasisTheory,
+  page: number
+): Promise<PaginatedList<ReactorFormula>> => {
+  const size = 5;
+
+  debug('Listing reactors', `page: ${page}`, `size: ${size}`);
```
Comment (djejaquino) [inline src/reactorFormulas/management.ts]:
```suggestion
  debug('Listing reactor formulas', `page: ${page}`, `size: ${size}`);
```

File: src/reactorFormulas/management.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  return selection;
+};
+
+const updateReactorFormula = (
+  bt: IBasisTheory,
+  id: string,
+  model: CreateReactorFormula
+): Promise<ReactorFormula> => {
+  debug(`Patching Reactor ${id}`, JSON.stringify(model, undefined, 2));
```
Comment (djejaquino) [inline src/reactorFormulas/management.ts]:
```suggestion
  debug(`Updating Reactor Formula ${id}`, JSON.stringify(model, undefined, 2));
```

File: src/reactorFormulas/utils.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  configuration,
+  requestParameters,
+}: CreateReactorFormula): CreateReactorFormulaModel {
+  return {
+    name,
+    description,
+    icon,
+    type,
+    code: code ? readFileContents(code) : '',
+    configuration: JSON.parse(readFileContents(configuration)),
```
Comment (djejaquino) [inline src/reactorFormulas/utils.ts:79]: Why not keep consistent with the env parser for configuration?

**Review by djejaquino — CHANGES_REQUESTED** [review]

File: src/reactorFormulas/utils.ts (hunk 5 above)
Comment (armsteadj1) [reply src/reactorFormulas/utils.ts:79]: I think the contract is different? in transforms you're actually setting the ENVs, this is just a definition of which ones you can pass in. So the `configuration` setting would actually be when you create Reactors.

Comment (djejaquino) [reply src/reactorFormulas/utils.ts:79]: You are right.

**Review by djejaquino — APPROVED** [review]

**Pull request merged**